runtime/lua/vim/diagnostic.lua

```diff
@@ -1564,16 +1564,23 @@ M.handlers.underline = {
 --- @param diagnostics table<integer, vim.Diagnostic[]>
 --- @param opts vim.diagnostic.Opts.VirtualText
 local function render_virtual_text(namespace, bufnr, diagnostics, opts)
+  -- Define the virtual text namespace if needed.
```

Comment:
I moved these blocks inside the specific render functions to allow external code to simply pass the diagnostic namespace, and not this one that we define inside `user_data` and is more of a private implementation detail.